src/components/EmailNotifications/Forms/KewordsForm.tsx

```diff
@@ -0,0 +1,116 @@
+import { INotification, useAddNotification, useEditNotification } from '@api';
+import { Input, FormControl, FormLabel, HStack, Button, Flex, useToast, Text } from '@chakra-ui/react';
+import { noop, parseAPIError } from '@utils';
+
+import { ChangeEvent, FormEvent, useState } from 'react';
+
+export const KeywordsForm = ({
+  onClose,
+  onUpdated = noop,
+  notification,
+}: {
+  onClose: () => void;
+  onUpdated?: () => void;
+  notification?: INotification;
+}) => {
+  const toast = useToast({ duration: 2000 });
+
+  const [keywords, setKeywords] = useState(notification?.data ?? '');
+
+  const [name, setName] = useState<string>(notification?.name ?? '');
+
+  const { mutate: addNotification, isLoading: isAdding } = useAddNotification();
+
+  const { mutate: editNofication, isLoading: isEditing } = useEditNotification();
+
+  const handleNameChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+  };
+
+  const handleKeywordsChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setKeywords(e.target.value);
+  };
+
+  const handleAddNotification = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!!notification) {
+      // edit existing
+      editNofication(
+        {
+          id: notification.id,
+          data: keywords,
+          name,
+        },
+        {
+          onSettled(data, error) {
+            if (error) {
+              toast({ status: 'error', title: 'Error', description: parseAPIError(error) });
```

Comment:
This should probably be handled slightly differently.  Currently the description is not actually getting parsed because it's path is different than the other microservices: (`response/data/msg`) instead of `message` -- so we see: 
![image](https://github.com/adsabs/nectar/assets/6970899/67850bf6-6ce6-492e-b5bf-3c65cb231fcb)
If I fix the parsing: 
![image](https://github.com/adsabs/nectar/assets/6970899/a272bed7-111d-4944-8f82-269f95480d43)

The error message is worse: 
![image](https://github.com/adsabs/nectar/assets/6970899/4a569cf7-6d47-4e08-a825-847ede491e5d)

I think, on error here we should probably check if it's syntax issue and alert on the form instead.

Reply:
@thostetler Please see my last commit. I added a syntax check.